include/rticonnextdds-connector.h

```diff
@@ -8,39 +8,194 @@
+
+int RTI_Connector_set_string_into_samples(
+	void *self,
+	const char *entity_name,
+	const char *name,
+	const char *value);
 
 int RTI_Connector_get_json_from_infos(
-        void *self,
-        const char *entity_name,
-        int index,
-        const char *name,
-        char **return_value);
+	void *self,
+	const char *entity_name,
+	int index,
+	const char *name,
+	char **value);
 
-int RTI_Connector_get_boolean_from_infos(
-        void *self,
-        int *return_value,
-        const char *entity_name,
-        int index,
-        const char *name);
-
-void* RTIDDSConnector_getJSONSample(void *connector, char *name, int index);
-double RTIDDSConnector_getNumberFromSamples(void *connector, char *name, int index, char *field_name);
-int RTIDDSConnector_getBooleanFromSamples(void *connector, char *name, int index, char *field_name);
-void* RTIDDSConnector_getStringFromSamples(void *connector, char *name, int index, char *field_name);
-char* RTIDDSConnector_freeString(char *json);
-int RTIDDSConnector_wait(void *connector, int timeout);
+int RTI_Connector_get_json_sample(
+	void *self,
+	const char *entity_name,
+	int index,
+	char **json_str);
+
+int RTI_Connector_get_json_member(
+	void *self,
+	const char *entity_name,
+	int index,
+	const char *member_name,
+	char **json_str);
+
+int RTI_Connector_clear(
+	void *self,
+	const char *entity_name);
+
+int RTI_Connector_read(
+	void *self,
+	const char *entity_name);
+
+int RTI_Connector_take(
+	void *self,
+	const char *entity_name);
+
+int RTI_Connector_write(
+	void *self,
+	const char *entity_name,
+	const char *params_json);
+
+struct RTI_Connector_Options {
+        /* boolean */ int enable_on_data_event;
+	/* boolean */ int one_based_sequence_indexing;
+};
+
+
+#define RTI_Connector_Options_INITIALIZER { \
+        1, /* enable_on_data_event */ \
+	1  /* one_based_sequence_indexing */ \
+}
+
+RTI_Connector *RTI_Connector_new(
+	const char *config_name,
+	const char *config_file,
+	const struct RTI_Connector_Options *options);
+
+void RTI_Connector_delete(RTI_Connector *self);
+
+int RTI_Connector_get_number_from_sample(
+	void *self,
+	double *return_value,
+	const char *entity_name,
+	int index,
+	const char *name);
+
+int RTI_Connector_get_boolean_from_sample(
+	void *self,
+	int *return_value,
+	const char *entity_name,
+	int index,
+	const char *name);
+
+int RTI_Connector_get_string_from_sample(
+	void *self,
+	char **return_value,
+	const char *entity_name,
+	int index,
+	const char *name);
+/*
+int RTI_Connector_get_any_from_sample(
+	void *self,
+	double *double_value_out,
+	RTIBool *bool_value_out,
+	char **string_value_out,
+	RTI_Connector_AnyValueKind *selected_out,
+	const char *entity_name,
+	int index,
+	const char *name);
+
+int RTI_Connector_get_any_from_info(
+	void *self,
+	double *double_value_out,
+	RTIBool *bool_value_out,
+	char **string_value_out,
+	RTI_Connector_AnyValueKind *selected_out,
+	const char *entity_name,
+	int index,
+	const char *name);
+ */
```

Comment:
What is this commented code? We should delete it if it is not in used.

Reply:
These are newly added functions. I will uncomment this once we implement the functions.

Reply:
Ok! Ideally we should try to avoid adding commented code without any comments. 😆